enumerables.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return true unless x != false
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return false unless yield(x) }
+    else
+      selfitem.my_each { |x| return false unless x }
+    end
+    true
+  end
```

Comment:
**[REQUIRED]**
* When no block or argument is given returns true when none of the collection members are false or nil.
```
Failure/Error: expect(false_array.my_all?).to be false_array.all?
     
       expected false
            got true
```
* When a class is passed as an argument returns true if all of the collection is a member of such class.
```
Failure/Error: expect(array.my_all?(Integer)).to be array.all?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns true if all of the collection matches the Regex.
```
Failure/Error: expect(words.my_all?(/d/)).to be words.all?(/d/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns true if all of the collection matches the pattern.
```
Failure/Error: expect(array.my_all?(3)).to be array.all?(3)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```